community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStore.java

```diff
@@ -900,12 +900,32 @@ public long nextCommittingTransactionId()
     @Override
     public void transactionCommitted( long transactionId, long checksum )
     {
-        if ( lastCommittedTx.offer( transactionId, checksum ) )
+        checkInitialized( lastCommittingTxField.get() );
+        TransactionId current = highestCommittedTransaction.get();
+        if ( current.transactionId() < transactionId )
```

Comment:
what kind of error will we get? I would expect one of the 2 to end up into the while loop, no?